sdk/core/azure-core/inc/uuid.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cstdlib> // for abort
+#include <new> // for placement new
+#include <string>
+#include <type_traits>
+#include <utility> // for swap and move
+
+namespace Azure { namespace Core {
+
+  class UUID {
+
+  private:
+    static const size_t UUIDSize = 16;
+
+    unsigned char m_uuid[UUIDSize];
+    // The UUID reserved variants.
+    static const unsigned char ReservedNCS = 0x80;
+    static const unsigned char ReservedRFC4122 = 0x40;
+    static const unsigned char ReservedMicrosoft = 0x20;
+    static const unsigned char ReservedFuture = 0x00;
+
+  public:
+    UUID()
+    {
+      std::random_device rd;
+      std::mt19937 gen(rd());
+      for (int i = 0; i < UUIDSize; i++)
+        m_uuid[i] = (unsigned char)gen();
+
+      //SetVariant to ReservedRFC4122
+      m_uuid[8] = (m_uuid[8] | ReservedRFC4122) & 0x7F;
+
+      const unsigned char version = 4;
+
+      m_uuid[6] = (m_uuid[6] & 0xF) | (version << 4);
+    }
+
+    std::string GetUUIDString() {
+      char s[36];
+
+      sprintf(&s[0], "%x%x%x%x-%x%x-%x%x-%x%x-%x%x%x%x%x%x",
```

Comment:
I am not sure why - we don't seem to include any C-style headers ('.h'), but I'd expect `sprintf` to be available only in the `std` namespace - i.e. only `std::sprintf()` should compile. Do we include `string.h` (instead of `cstring`) anywhere?